### DISCUSSION TITLE
Should there be beta-like process for all translations?

### DISCUSSION THREAD

**Discussion by XRFXLP** [codewars/codewars.com#2382] [Ideas]
Should there be a beta-like process for all translations with a relatively lenient condition of approval?

There are numbers of benefits in doing so, couple of them are following:
- Translation would get more feedback, would be idiomatic, of good quality consistent with other languages, and unsuited tasks for a particular language can be avoided, for instance doing mathematical katas with bash script ( actually there are worse examples than this ).
and it is not uncommon to find author themselves publishing translations which is of really bad quality, or author happily accepting broken translations.

- Translations would be visible to everyone, people can choose to filter translations by language (and at least I believe this would be a genuine improvement on CW) and hence would get faster response and keeping people motivated to write a translation.

  Right now they're only visible for some selected users, although doing so might've been the easiest option available at that time, to me it sounds biased and unreasonable, as the major factor in deciding this was trust, and trust is primarily made up from the actions like sticking to the platform and behave appropriately for a reasonably long period of time, but unfortunately, this has nothing to do with language proficiency and in some cases, it is quite the opposite.

**Comment by hobovsky** [comment]
For those who missed it, [post by kazk](https://github.com/codewars/codewars.com/issues/2372#issuecomment-832184985):

> From internal chat:
> 
> > My current plan for the huge list of pending translations is to revamp the translation approval process. I think we have the following problems at least:
> > 
> > 1. Discoverability: Difficult to find pending translations to begin with. Fixing this is easy, but that can degrade the quality of approved translations.
> > 2. Scalability: The current workflow doesn't scale well. We don't have many reviewers and reviewing is time-consuming. Fixing discoverability can increase the number, but can degrade the quality.
> > 3. Workflow: The current workflow is terrible. Too much responsibility for a single reviewer. Reviewing is difficult as well because you'll need to fork it and try some solutions. Merging the translation is also problematic.
> > 
> > Possible solution:
> > 
> > * Make the translation completable. Let users solve like kata and leave feedback. They'll get rewarded for completing and giving feedback like beta kata.
> > * Make them only approvable after collecting multiple approval votes. I think this vote will be different from kata approval. It's more like agree to approve or not.
> > * Make a page similar to kata search for this.
> > * Make merge conflict easy to fix. It should be enough to show the description with conflict markers when approving. Then making the approver fix just before merging.
> > * Make kata easier to fix. Allow forking the current state to propose fix. It should be easy to fix any mistakes.
> > * Allow mods to retire/pause language. As a last resort, we should be able to disable further completions.

**Comment by Kacarott** [comment]
Regarding implementation, one idea would be a two-stage process for translation approval.
1. First stage would be pretty much exactly what we have now. A translation is published, and a reviewer reviews it and presses "Approve as beta"
2. The translation is now in a "Beta" state.
  - It is still select-able from the main kata page, but would be clearly indicated with a beta tag. 
  - On completion of a beta translation, options for review would be similar to a beta kata (asks for satisfaction, and if the difficulty is appropriate for the kata rank -> "Too easy", "Appropriate", "Too hard")
  - Potentially, a beta translation could also have a dedicated discourse page too for discussing issues
3. Once there are enough solves, the ratings are satisfactory, and no issues (specific to the translation) it could be automatically approved.
4. In the case that there are too many issues, or it is decided that the translation is not appropriate, it could be retired, removing it from the kata.

I also imagine that a search filter to include/exclude beta translations would be prudent.
